### PR TITLE
Feature/start weekday

### PR DESCRIPTION
**Pull request by sachinatcodewalnut**
# PR Template & Definition of Done

## What does this PR do?

-
-
-

## What steps does your reviewer have to take to test this PR manually?

1.
2.
3.

## Testing checklist - Please check off

- [ ] I have written unit tests for all components files I've worked on, these tests cover what the user sees and how the UI changes when they interact with it.
- [ ] I have written unit tests for all helper functions I've worked on, that cover use-cases and edge-cases.
- [ ] I have written integration tests for features I've worked on, to ensure my components work together as I expect.

### If you have not added tests for any of the above, please explain why below.

...

## Definition of Done - Please check off

- [ ] I have completed the above testing checklist so my code is well tested and I have confidence my code works as I expect in a variety of situations.
- [ ] I have Eslint and Prettier enabled to run on file save and I have fixed all errors highlighted by Eslint.
- [ ] I have added complete Prop Types and type definitions without using `any` for all my components and helper functions.
- [ ] I have deleted all non-descriptive comments and dead code from the files I've touched.
- [ ] I have rebased my branch with the base branch I want to merge into and all the commits in this PR are my own.

### CODE REVIEW

**Review by Sarath-Kumar-G — DISMISSED** [review]

**Comment by Sarath-Kumar-G** [comment]
`git push <branch name> --force-with-lease` can be performed to avoid commit duplication in the rebase

**Comment by Sarath-Kumar-G** [comment]
Conventional commits are missing.

File: src/components/atoms/DatePicker/DatePicker.tsx [hunk 1]
```diff
@@ -39,7 +53,7 @@ export function CustomDatePicker({
   return (
     <div className={datePickerDynamicClasses}>
       <span className="text-sm font-semibold	 whitespace-nowrap absolute -top-3 left-3 px-1 bg-white text-black text-opacity-60">
-        Start Date
+        {startDateHeading}
```
Comment (Sarath-Kumar-G) [inline src/components/atoms/DatePicker/DatePicker.tsx:56]: it can be `startDateLabel`

**Comment by Sarath-Kumar-G** [comment]
image
@sachinatcodewalnut Please check these error suggestions. The build is failing. PFA

File: src/components/atoms/DatePicker/DatePicker.test.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
+import React from "react";
+import { render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
```
Comment (Sarath-Kumar-G) [inline src/components/atoms/DatePicker/DatePicker.test.tsx:3]: We can remove unused code.

File: src/components/atoms/DatePicker/DatePicker.tsx [hunk 3]
```diff
@@ -4,14 +4,28 @@ import DatePicker from "react-datepicker";
...
+  calendarStartFromMondayDay,
+  startDatePlaceHolder,
+  startDateHeading,
+  endDateHeading,
+  endDatePlaceHolder,
+  endDateLimit,
+  markSundayAsRed
 }: CustomDatePickerProps) {
   const [startDate, setStartDate] = useState<Date>(new Date());
   const [isStartDateOpen, setIsStartDateOpen] = useState<boolean>(false);
```
Comment (Sarath-Kumar-G) [inline src/components/atoms/DatePicker/DatePicker.tsx:31]: `isStartDateOpen` this variable is not in use, we can remove if this state is not needed.

Comment (Sarath-Kumar-G) [reply src/components/atoms/DatePicker/DatePicker.tsx:31]: Same for `isEndDateOpen`

**Review by Sarath-Kumar-G — CHANGES_REQUESTED** [review]

File: src/components/molecules/MyComponent/MyComponent.tsx [hunk 4]
```diff
@@ -1,9 +1,17 @@
...
+  endDateLimit.setDate(endDateLimit.getDate() + 30);
   return (
     <div className="w-full flex justify-center">
-      <CustomDatePicker isEndDate />
+      <CustomDatePicker isEndDate 
+                        startDatePlaceHolder="Start Date"
+                        endDatePlaceHolder="End Date"
+                        calendarStartFromMondayDay={false}
+                        endDateLimit={endDateLimit}
+                        markSundayAsRed={true}
```
Comment (Sarath-Kumar-G) [inline src/components/molecules/MyComponent/MyComponent.tsx:13]: It just `markSundayAsRed`, it's read as `true` flag.

File: src/components/molecules/MyComponent/MyComponent.tsx [hunk 5]
```diff
@@ -1,9 +1,17 @@
...
 export function MyComponent() {
+  const endDateLimit = new Date();
+  endDateLimit.setDate(endDateLimit.getDate() + 30);
   return (
     <div className="w-full flex justify-center">
-      <CustomDatePicker isEndDate />
+      <CustomDatePicker isEndDate 
+                        startDatePlaceHolder="Start Date"
+                        endDatePlaceHolder="End Date"
+                        calendarStartFromMondayDay={false}
```
Comment (Sarath-Kumar-G) [inline src/components/molecules/MyComponent/MyComponent.tsx:11]: This can be the same as a `true` flag if we won't provide props it'll take `false`.

Comment (Sarath-Kumar-G) [reply src/components/molecules/MyComponent/MyComponent.tsx:11]: Please read esLint suggestions for examples.
```JavaScript
 <CustomDatePicker isEndDate 
                        startDatePlaceHolder="Start Date"
                        endDatePlaceHolder="End Date"
                        //calendarStartFromMondayDay={false} => It'll be false
                        endDateLimit={endDateLimit}
                        markSundayAsRed // It'll be true
                          />

**Review by Sarath-Kumar-G — CHANGES_REQUESTED** [review]

File: src/components/atoms/DatePicker/DatePicker.tsx (hunk 1 above)
Comment (sachinatcodewalnut) [reply src/components/atoms/DatePicker/DatePicker.tsx:55]: updated;

File: src/components/atoms/DatePicker/DatePicker.tsx (hunk 3 above)
Comment (sachinatcodewalnut) [reply src/components/atoms/DatePicker/DatePicker.tsx:31]: updated;

File: src/components/molecules/MyComponent/MyComponent.tsx (hunk 4 above)
Comment (sachinatcodewalnut) [reply src/components/molecules/MyComponent/MyComponent.tsx:13]: updated;

File: src/components/molecules/MyComponent/MyComponent.tsx (hunk 5 above)
Comment (sachinatcodewalnut) [reply src/components/molecules/MyComponent/MyComponent.tsx:11]: got it; thanks;

**Review by Sarath-Kumar-G — APPROVED** [review]
